File: backend/internal/service/system/admin/v1/base_table_backup.go
package admin

import (
	"context"
	"fmt"

	"github.com/go-kratos/kratos/v3/log"
	adminv1 "github.com/liujitcn/kratos-admin/backend/api/gen/go/system/admin/v1"
	biz "github.com/liujitcn/kratos-admin/backend/internal/biz/system/admin"
	"github.com/liujitcn/kratos-core/errorsx"
	"google.golang.org/protobuf/types/known/emptypb"
)

// BaseTableBackupService 管理数据库备份配置。
type BaseTableBackupService struct {
	adminv1.UnimplementedBaseTableBackupServiceServer
	baseTableBackupCase *biz.BaseTableBackupCase
}

// NewBaseTableBackupService 创建数据库备份配置服务。
func NewBaseTableBackupService(baseTableBackupCase *biz.BaseTableBackupCase) *BaseTableBackupService {
	return &BaseTableBackupService{baseTableBackupCase: baseTableBackupCase}
}

// PageBaseTableBackup 分页查询数据库备份配置。
func (s *BaseTableBackupService) PageBaseTableBackup(ctx context.Context, req *adminv1.PageBaseTableBackupRequest) (*adminv1.PageBaseTableBackupResponse, error) {
	result, err := s.baseTableBackupCase.PageBaseTableBackup(ctx, req)
	if err != nil {
		log.Error(fmt.Sprintf("PageBaseTableBackup %v", err))
		return nil, errorsx.WrapInternal(err, "查询备份配置失败")
	}
	return result, nil
}

// GetBaseTableBackup 查询数据库备份配置。
func (s *BaseTableBackupService) GetBaseTableBackup(ctx context.Context, req *adminv1.GetBaseTableBackupRequest) (*adminv1.BaseTableBackupForm, error) {
	result, err := s.baseTableBackupCase.GetBaseTableBackup(ctx, req.GetId())
	if err != nil {
		log.Error(fmt.Sprintf("GetBaseTableBackup %v", err))
		return nil, errorsx.WrapInternal(err, "查询备份配置失败")
	}
	return result, nil
}

// CreateBaseTableBackup 创建数据库备份配置。
func (s *BaseTableBackupService) CreateBaseTableBackup(ctx context.Context, req *adminv1.CreateBaseTableBackupRequest) (*emptypb.Empty, error) {
	err := s.baseTableBackupCase.CreateBaseTableBackup(ctx, req.GetBaseTableBackup())
	if err != nil {
		log.Error(fmt.Sprintf("CreateBaseTableBackup %v", err))
		return nil, errorsx.WrapInternal(err, "创建备份配置失败")
	}
	return new(emptypb.Empty), nil
}

// UpdateBaseTableBackup 更新数据库备份配置。
func (s *BaseTableBackupService) UpdateBaseTableBackup(ctx context.Context, req *adminv1.UpdateBaseTableBackupRequest) (*emptypb.Empty, error) {
	err := s.baseTableBackupCase.UpdateBaseTableBackup(ctx, req.GetBaseTableBackup())
	if err != nil {
		log.Error(fmt.Sprintf("UpdateBaseTableBackup %v", err))
		return nil, errorsx.WrapInternal(err, "更新备份配置失败")
	}
	return new(emptypb.Empty), nil
}

// DeleteBaseTableBackup 删除数据库备份配置。
func (s *BaseTableBackupService) DeleteBaseTableBackup(ctx context.Context, req *adminv1.DeleteBaseTableBackupRequest) (*emptypb.Empty, error) {
	err := s.baseTableBackupCase.DeleteBaseTableBackup(ctx, req.GetId())
	if err != nil {
		log.Error(fmt.Sprintf("DeleteBaseTableBackup %v", err))
		return nil, errorsx.WrapInternal(err, "删除备份配置失败")
	}
	return new(emptypb.Empty), nil
}

// SetBaseTableBackupStatus 设置数据库备份配置状态。
func (s *BaseTableBackupService) SetBaseTableBackupStatus(ctx context.Context, req *adminv1.SetBaseTableBackupStatusRequest) (*emptypb.Empty, error) {
	err := s.baseTableBackupCase.SetBaseTableBackupStatus(ctx, req)
	if err != nil {
		log.Error(fmt.Sprintf("SetBaseTableBackupStatus %v", err))
		return nil, errorsx.WrapInternal(err, "设置备份配置状态失败")
	}
	return new(emptypb.Empty), nil
}

File: frontend/admin/packages/modules/system/src/api/system/admin/v1/base_table_backup.ts
import service from "@liujitcn/kratos-admin-core/request";
import type { Empty } from "@liujitcn/kratos-admin-system/rpc/google/protobuf/empty";
import type { BaseTableBackupService } from "@liujitcn/kratos-admin-system/rpc/system/admin/v1/base_table_backup";
import type {
  BaseTableBackupForm,
  CreateBaseTableBackupRequest,
  DeleteBaseTableBackupRequest,
  GetBaseTableBackupRequest,
  PageBaseTableBackupRequest,
  PageBaseTableBackupResponse,
  SetBaseTableBackupStatusRequest,
  UpdateBaseTableBackupRequest
} from "@liujitcn/kratos-admin-system/rpc/system/admin/v1/base_table_backup";

const BASE_TABLE_BACKUP_URL = "/v1/admin/base/table-backup";

/** 数据库备份配置服务实现。 */
export class BaseTableBackupServiceImpl implements BaseTableBackupService {
  /** 分页查询备份配置。 */
  PageBaseTableBackup(request: PageBaseTableBackupRequest): Promise<PageBaseTableBackupResponse> {
    return service<PageBaseTableBackupRequest, PageBaseTableBackupResponse>({
      url: BASE_TABLE_BACKUP_URL,
      method: "get",
      params: request
    });
  }

  /** 查询备份配置。 */
  GetBaseTableBackup(request: GetBaseTableBackupRequest): Promise<BaseTableBackupForm> {
    return service<GetBaseTableBackupRequest, BaseTableBackupForm>({
      url: `${BASE_TABLE_BACKUP_URL}/${request.id}`,
      method: "get"
    });
  }

  /** 创建备份配置。 */
  CreateBaseTableBackup(request: CreateBaseTableBackupRequest): Promise<Empty> {
    return service<BaseTableBackupForm | undefined, Empty>({
      url: BASE_TABLE_BACKUP_URL,
      method: "post",
      data: request.base_table_backup
    });
  }

  /** 更新备份配置。 */
  UpdateBaseTableBackup(request: UpdateBaseTableBackupRequest): Promise<Empty> {
    return service<BaseTableBackupForm | undefined, Empty>({
      url: `${BASE_TABLE_BACKUP_URL}/${request.base_table_backup?.id ?? ""}`,
      method: "put",
      data: request.base_table_backup
    });
  }

  /** 删除备份配置。 */
  DeleteBaseTableBackup(request: DeleteBaseTableBackupRequest): Promise<Empty> {
    return service<DeleteBaseTableBackupRequest, Empty>({ url: `${BASE_TABLE_BACKUP_URL}/${request.id}`, method: "delete" });
  }

  /** 设置备份配置状态。 */
  SetBaseTableBackupStatus(request: SetBaseTableBackupStatusRequest): Promise<Empty> {
    return service<SetBaseTableBackupStatusRequest, Empty>({
      url: `${BASE_TABLE_BACKUP_URL}/${request.id}/status`,
      method: "put",
      data: request
    });
  }
}

/** 数据库备份配置服务实例。 */
export const defBaseTableBackupService = new BaseTableBackupServiceImpl();
